### PR TITLE
add MSstatsTMT and update MSstats

### PR DESCRIPTION
**Pull request by daichengxin**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/quantms/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/quantms _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -154,10 +154,26 @@ params {
     max_fr_mz       = null
     diann_normalize = true
 
+    // MSstats and MSstatsTMT shared
+    msstats_tmt_remove_one_feat_prot    = true
```
Comment (jpfeuffer) [inline nextflow.config:158]: I think we should either remove the msstats prefix for all parameters (and group them nicely in the param.json)
or we should use a consistent prefix:
msstats for general, msstatslfq for lfq specific and msstatstmt/iso for iso.

File: nextflow.config [hunk 2]
```diff
@@ -154,10 +154,26 @@ params {
...
+    ref_condition                       = null
+    contrasts                           = 'pairwise'
+
     // MSstats
-    skip_post_msstats = false
-    ref_condition     = null
-    contrasts         = 'pairwise'
+    skip_post_msstats                           = false
```
Comment (jpfeuffer) [inline nextflow.config:163]: we dont need two skip parameters unless you want to support mixed designs in the near future.

File: nextflow.config [hunk 3]
```diff
@@ -154,10 +154,26 @@ params {
...
+    msstats_keep_feats_with_few_measurements    = false
```
Comment (jpfeuffer) [inline nextflow.config:164]: Can't you unify this with `msstatstmt_rmpsm_withfewmea_withinrun`.
Should be the same, no?
What about the summary method, do we need two summary method parameters?

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

**Comment by ypriverol** [comment]
@daichengxin MSstats is failing with DIANN

**Comment by daichengxin** [comment]
ok. I am testing locally.

**Comment by ypriverol** [comment]
I think is because in the last tests we have included only two files in the DIANN with no multiple conditions. But this is only a guess.

File: nextflow.config (hunk 3 above)
Comment (daichengxin) [reply nextflow.config:164]: There is a difference between these two parameters. One is across run. The other is in withinrun.  For summary method. LFQ options include TMP and `linear`. But ISO includes `msstats` `medianPolish` `Median` `LogSum`.

**Comment by daichengxin** [comment]
Yes. you are right. I use two conditions locally to test. Need to update the test data

File: nextflow.config (hunk 2 above)
Comment (daichengxin) [reply nextflow.config]: done

**Comment by daichengxin** [comment]
The test passed. Need to upload the latest dia test data to nfcore, then recover test_dia config. https://raw.githubusercontent.com/daichengxin/quantms/dev/assets/PXD026600.sdrf.tsv

**Comment by jpfeuffer** [comment]
Why are you not using the newest version with argument parsing? Because of the additional dependency?
https://github.com/nf-core/proteomicslfq/blob/dev/bin/msstats_plfq.R

**Comment by daichengxin** [comment]
Yes. `argparse` is not in the container

File: bin/msstats_plfq.R [hunk 4]
```diff
@@ -1,68 +1,128 @@
...
 
-usage <- "Rscript msstats_plfq.R input.csv input.mztab [list of contrasts or 'pairwise'] [default control condition or ''] [output prefix]"
-
-#TODO Waiting DIA mzTab
-if (length(args)<2) {
+#TODO rewrite mzTab in next version
+if (length(args)<1) {
     print(usage)
-    stop("At least the first two arguments must be supplied (input csv and input mzTab).n", call.=FALSE)
+    stop("At least the first one argument must be supplied (input csv).n", call.=FALSE)
```
Comment (jpfeuffer) [inline bin/msstats_plfq.R:9]:
```suggestion
    stop("At least the first argument must be supplied (input csv).n", call.=FALSE)
```

File: bin/msstats_tmt.R [hunk 5]
```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env Rscript
+args = commandArgs(trailingOnly=TRUE)
+char_to_boolean = c("true"=TRUE, "false"=FALSE)
+
+usage <- "Rscript msstats_tmt.R input.csv [list of contrasts or 'pairwise'] [default control condition or '']... [normalization based reference channel]"
+if (length(args)<1) {
+    print(usage)
+    stop("At least the first one argument must be supplied (input csv).n", call.=FALSE)
```
Comment (jpfeuffer) [inline bin/msstats_tmt.R:8]:
```suggestion
    stop("At least the first argument must be supplied (input csv).n", call.=FALSE)
```

**Comment by jpfeuffer** [comment]
Can we maybe have the logging changes separate? I don't think this is ready for release.

**Comment by jpfeuffer** [comment]
E.g. it needs a lot of testing, the testdata needs to be on nf-core, I think it is better to only have one MSstats step (we need to check whether there are more parameters that we can combine).

**Comment by daichengxin** [comment]
So do we need to remove the logging changes in this PR?

**Comment by jpfeuffer** [comment]
No but create a PR with just the logging changes.

**Comment by jpfeuffer** [comment]
Use `git cherry-pick` on a new branch

**Comment by ypriverol** [comment]
Can I merge this PR @jpfeuffer ?

**Comment by ypriverol** [comment]
We need two different PRs? Why we can combine both changes here if they are correct.

**Comment by ypriverol** [comment]
But @daichengxin changed the MSstats as requested, then both things are OK now.

**Comment by jpfeuffer** [comment]
> E.g. it needs a lot of testing, the testdata needs to be on nf-core, I think it is better to only have one MSstats step/script (we need to check whether there are more parameters that we can combine).

**Comment by jpfeuffer** [comment]
You can contribute to this PR. I just want to merge into master soon, and do not want to have half-ready PRs in there.

**Comment by daichengxin** [comment]
I got the changes that needs to be done.  But I'm afraid I won't be able to do it soon because I'm not very familiar with the mechanics of CI.

**Pull request merged**